crates/runc-shim/src/common.rs

```diff
@@ -93,13 +92,15 @@ pub fn create_io(
             stdio.stdout.as_str(),
             stdio.stderr.as_str()
         );
-        let io = FIFO {
-            stdin: stdio.stdin.to_string().none_if(|x| x.is_empty()),
-            stdout: stdio.stdout.to_string().none_if(|x| x.is_empty()),
-            stderr: stdio.stderr.to_string().none_if(|x| x.is_empty()),
+        let opts = IOOption {
+            open_stdin: !stdio.stdin.is_empty(),
+            open_stdout: !stdio.stdout.is_empty(),
+            open_stderr: !stdio.stderr.is_empty(),
         };
+        let io = PipedIo::new(_io_uid, _io_gid, &opts).unwrap();
+
         pio.io = Some(Arc::new(io));
-        pio.copy = false;
```

Comment:
Instead of piping io, we have a different direct IO solution: https://github.com/containerd/rust-extensions/pull/24#issuecomment-1040619960.